feat: adds basic instructions for nvim-lspconfig;

megalithic opened the pull request:
- now that nvim-lspconfig has a basic config for `next-ls`; it's easier than ever to maintain your own next-ls install in neovim. should you choose to not using `elixir-tools.nvim` for a more full-featured config/setup.

megalithic committed:
feat: adds basic instructions for nvim-lspconfig;

- now that nvim-lspconfig has a basic config for `next-ls`; it's easier
  than ever to maintain your own next-ls install in neovim. should you
  choose to not using `elixir-tools.nvim` for a more full-featured
  config/setup.

mhanberg commented on _pages/docs/next-ls/editors.md:
````diff
@@ -14,11 +14,17 @@ github_stars: elixir-tools/next-ls
 
 ## Neovim
 
-[elixir-tools.nvim](https://github.com/elixir-tools/elixir-tools.nvim) is a first party elixir-tools editor plugin and will install and manage Next LS for you
+- [elixir-tools.nvim](https://github.com/elixir-tools/elixir-tools.nvim) is a first party elixir-tools editor plugin and will install and manage Next LS for you
+- You might also decide to install and manage `next-ls` yourself using `nvim-lspconfig`, and perhaps, [`homebrew`](/docs/next-ls/installation/#homebrew) for the `nextls` binary.
+  - ```lua
+    require("lspconfig")["nextls"].setup({
+      cmd = {"nextls", "--stdio"} -- assumes you have `nextls` in your `$PATH`
+    })
+    ```
 
````
instead of a bulleted list, lets use h3 sections for elixir-tools.nvim and nvim-lspconfig, similar to how the emacs sections are.

this also allows you to deep link to each section

mhanberg replied:
Also, if you could add the example of how to configure the initialization options, that would be great too

megalithic committed:
fix: addresses some PR comments

- formatting/presentation of sections
- adds `init_options` table to config for completions support

mhanberg commented on _pages/docs/next-ls/editors.md:
```diff
@@ -14,11 +14,30 @@ github_stars: elixir-tools/next-ls
+
 [elixir-tools.nvim](https://github.com/elixir-tools/elixir-tools.nvim) is a first party elixir-tools editor plugin and will install and manage Next LS for you
 
+### [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#nextls)
```
Remove the links from the headers. There are already links in the body of each section.

mhanberg commented on _pages/docs/next-ls/editors.md:
````diff
@@ -14,11 +14,30 @@ github_stars: elixir-tools/next-ls
+
+```lua
+ require("lspconfig")["nextls"].setup({
+   cmd = {"nextls", "--stdio"} -- assumes you have the executable `nextls` in your `$PATH`
````
Missing a comma 

I don't think you need the comments here.

megalithic committed:
fix: add comma; remove comments in config sample

megalithic committed:
fix: remove links from headers

mhanberg committed:
fixup! feat: adds basic instructions for nvim-lspconfig;

mhanberg commented:
Thanks!
